packages/js/components/src/select-control/types.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { ReactElement, Component } from 'react';
+import {
+	UseComboboxGetItemPropsOptions,
+	UseComboboxGetMenuPropsOptions,
+	GetPropsCommonOptions,
+} from 'downshift';
+
+export type ItemType = {
+	value: string;
+	label: string;
+};
```

Comment:
Could we expose this type as part of the `@woocommerce/components` package. `ItemType` would be to generic, so maybe something like `SelectControlItem`?

Also it would be nice to have an extra optional param with extra data (this is not a must). But in our instance where we are displaying a tree like structure, it be nice to add the children as part of this data.
But I can also work around this, by keeping a key/value store around and making use of this in my own component.

Reply:
We can expose this in the follow-up.  I'm not happy with exposing the type as it is currently since it doesn't have to follow this format at all.

Reply:
Ooh really, I thought it had to follow this format in order for the selected items to show up correctly?
If any type can be passed in, we should add support for a generic type.

Reply:
Technically, any type can be passed in since you can pass your own `itemToString` method.

But agreed, it would be good to have a type for this that's exported.  I'd still love to handle this in a follow-up if possible to unblock other work on this PR, but if you feel strongly about implementing it first here, we can.